### PR TITLE
Rename RKernel SDK modules

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR:
- Splits `instance_attestation` into `attestation` and `crypto` modules
  - Because `attestation` module will be removed once we have an Orchestrator for RKernel
  - And `crypto` module will contain the logic for both Instance and Group keys
- Renames `mock_attestation` to `testing` since it not only contains the attestation logic but also the crypto logic

### CODE REVIEW

File: oak_restricted_kernel_sdk/src/mock_sdk.rs [hunk 1]
```diff
@@ -34,7 +37,7 @@ lazy_static::lazy_static! {
   };
 }
 
-#[cfg(feature = "mock_attestation")]
+#[cfg(feature = "mock_sdk")]
```
Comment (tiziano88) [inline oak_restricted_kernel_sdk/src/mock_sdk.rs:40]: Isn't this entire mod already feature gated?

**Comment by ipetr0v** [comment]
> What is the migration plan for our clients?

Internal clients depend on a specific commit. So once we submit this PR, we would be able to update their code that reference them to the new commit

File: oak_restricted_kernel_sdk/src/mock_sdk.rs (hunk 1 above)
Comment (ipetr0v) [reply oak_restricted_kernel_sdk/src/mock_sdk.rs]: Removed

**Review by tiziano88 — APPROVED** [review]
LG, but please for @jul-sh to approve

**Comment by jul-sh** [comment]
I don't have super strong objection to merging this. 

But the rationale for the original naming choice of `attestation` that the provided crypto functions operate with keys that are specifically bound to the attestation. — Even when we remove the struct get the evidence itself, the signing and encryption keys are still bound to it.

@ipetr0v How would you want to handle group attestation if we support it in the future? The current mod was called `instance_attestation`, so we'd have the option of additionally offering `group_attestation` in the future.

**Comment by ipetr0v** [comment]
> I don't have a strong objection to merging this.
> 
> But the rationale for the original naming choice of `attestation` that the provided crypto functions operate with keys that are specifically bound to the attestation. — Even if we remove the struct get the evidence itself, the signing and encryption keys are still bound to it.
> 
> @ipetr0v How would you want to handle group attestation if we support it in the future? The current mod was called `instance_attestation`, so we'd have the option of additionally offering an additional `group_attestation` mod in the future.

There will be no group attestation, only group encryption. So `GroupEvidence` will not be there. And `GroupKeyHandle` will be in the `crypto` module.

Also I think crypto is not a part of the attestation. It's definitely bound to it, but it's a separate logic on top of this.

**Review by jul-sh — APPROVED** [review]

**Pull request merged**